source/dofs/dof_renumbering.cc

```diff
@@ -1863,7 +1857,20 @@ namespace DoFRenumbering
     for (unsigned int i=0; i<n_dofs; ++i)
       new_indices[i] = i;
 
-    std::random_shuffle (new_indices.begin(), new_indices.end());
+    // shuffle the elements; the following is essentially the
+    // std::random_shuffle algorithm but uses a predictable
+    // random number generator
+    ::boost::mt19937 random_number_generator;
+    for (unsigned int i=1; i<n_dofs; ++i)
```

Comment:
interesting that the example implementations at [1] all loop backwards. I assume they are statistically equivalent, though. Starting at 1 instead of 0 is important, though.

[1]: http://en.cppreference.com/w/cpp/algorithm/random_shuffle